### ISSUE TITLE
Updated MML model file?

### ISSUE THREAD

**Issue by dhairyadalal** [allenai/allennlp#2598]
Is there an updated mml model file for the wikitables parser? I tried to load "wikitables-model-2018.09.14 (1).tar.gz"  from https://github.com/allenai/allennlp/issues/1816 and got the following error:

```
from allennlp.models.archival import load_archive
from allennlp.predictors import Predictor
archive = load_archive("wikitables-model-2018.09.14 (1).tar.gz")
```
```
RuntimeError: Error(s) in loading state_dict for WikiTablesMmlSemanticParser:
	Missing key(s) in state_dict: "_type_params.weight", "_type_params.bias", "_decoder_step._entity_type_embedding.weight". 
	Unexpected key(s) in state_dict: "_entity_type_encoder_embedding.weight", "_entity_type_decoder_embedding.weight". 
```

I'd like to trying running the wikitables semantic parser demo locally.

**Comment by matt-gardner** [comment]
The demo runs on current code, and it's using the model that's found here: https://github.com/allenai/allennlp-demo/blob/ffe7c7892eebf5e2c4c797f2034ce2ca0c058589/server/models.py#L86

So downloading that model _should_ work if you have current code.

That model's not particularly great, though, and we have a much better one (from a paper that just got accepted at NAACL), it's just not merged into master yet.  @pdasigi, any ETA for when we might have that model merged here?

**Comment by dhairyadalal** [comment]
Thanks @matt-gardner. It looks like we were running version 0.5. After upgrading it, we got it to work. So looking through the history, it seems like the model performance is limited due to limited training. Do you imagine the model would get better with more training examples? 

Also we're exploring the semantic parsing framework to support generalized database queries against arbitrary single tables. For our training situation, we would not have the ability to generate logical forms for supervised training. Wondering if we could set up a weakly supervised (http://www.aclweb.org/anthology/Q13-1005) solution with the framework using the denotation as our training signal. This is a bit of a new space for me, so I'm entirely sure if this would be ideal or even is a good idea. Appreciate your help .

**Comment by matt-gardner** [comment]
This is exactly the setting that @pdasigi tackled in our recent NAACL paper.  I'll let him give you more information here.

**Comment by pdasigi** [comment]
@dhairyadalal Regarding your original question, I'm trying to get the latest MML model out in a week. 🤞

Regarding model performance, I think the amount of training data is definitely a limitation, especially given that the supervision comes from denotations alone. We do not have a good way of generating logical forms that generalize to unseen tables given the current data yet. Our NAACL paper addresses this problem to some extent, but there's still a lot of room for improvement.

Regarding dealing with weak supervision, we took an iterative approach to training semantic parsers in our paper, where we train a model with denotation supervision that searches for logical forms of limited complexity (or length) during training, decode from it to generate data to train an MML model, and then use that MML model to initialize weights of the model that can search for logical forms till a greater length. Our paper shows that such an iterative approach provides curriculum-like guidance during training, and yields high quality models. We'll have the paper up on arXiv soon (hopefully in a week or so).

In case you want to take a look at some code, [this class](https://github.com/allenai/allennlp/blob/master/allennlp/models/semantic_parsing/wikitables/wikitables_erm_semantic_parser.py) is an older version of the model for WikiTables that searches during training with denotation supervision. The core training logic is all in there.

In short, I think there are ways to train a weakly supervised parser for your case. I can give you more pointers if you can tell me specific details of your problem.

**Comment by dhairyadalal** [comment]
thanks for sharing. I'll follow-up offline @pdasigi. Look forward to reading your NAACL paper.
